contracts/Delegations.sol

```diff
@@ -0,0 +1,130 @@
+pragma solidity 0.5.16;
+
+import "@openzeppelin/contracts/math/SafeMath.sol";
+import "@openzeppelin/contracts/math/Math.sol";
+
+import "./spec_interfaces/ICommitteeListener.sol";
+import "./interfaces/IElections.sol";
+import "./spec_interfaces/IValidatorsRegistration.sol";
+import "./IStakingContract.sol";
+import "./spec_interfaces/ICommittee.sol";
+import "./spec_interfaces/ICompliance.sol";
+import "./ContractRegistryAccessor.sol";
+import "./spec_interfaces/IDelegation.sol";
+
+contract Delegations is IDelegations, IStakeChangeNotifier, ContractRegistryAccessor {
+	using SafeMath for uint256;
+
+	// TODO consider using structs instead of multiple mappings
+	mapping (address => uint256) uncappedStakes;
+	uint256 totalGovernanceStake; // TODO - move to elections
+
+	mapping (address => address) delegations;
+
+	modifier onlyStakingContract() {
+		require(msg.sender == address(getStakingContract()), "caller is not the staking contract");
+
+		_;
+	}
+
+	constructor() public {
+	}
+
+	function delegate(address to) external {
+		address prevDelegatee = getDelegation(msg.sender);
+
+		uint256 prevGovStakePrevDelegatee = getGovernanceEffectiveStake(prevDelegatee);
+		uint256 prevGovStakeNewDelegatee = getGovernanceEffectiveStake(to);
+
+		delegations[msg.sender] = to; // delegation!
+
+		uint256 delegatorStake = getStakingContract().getStakeBalanceOf(msg.sender);
+
+		uint256 newStakePrevDelegatee = uncappedStakes[prevDelegatee].sub(delegatorStake);
```

Comment:
not enough meat, remember that totalGovernanceStake is going to Elections... 

we are left with just one line which updates uncapped, which is also not similar to applyStakeChangeLocally since it works with _sign and here in delegate we do not